frontend/hooks/notifcation.ts

```diff
@@ -0,0 +1,33 @@
+export function askNotificationPermissions() {
+    if (!("Notification" in window)) {
+        console.log("This browser does not support desktop notification");
+    } else {
+        try {
+            Notification.requestPermission();
+        } catch {
+            console.log("Notifications API not supported on this device");
+        }
+    }
+}
+
+export function playNotification(message: string) {
+    if ("Notification" in window) {
+        try {
+            /* eslint-disable-next-line */
+            new Notification("Alert", {
+                body: message,
+                icon: "../favicon.ico",
+            });
+        } catch (e) {
+            console.log("Notifications API not supported on this device");
+        }
+    }
+}
+
+export function checkPermissions() {
+    return localStorage && localStorage.getItem("notifs") === "false";
+}
+
+export function getPermissions() {
```

Comment:
Not exactly sure what the return type will be - I think it might be bool | string? Perhaps it makes sense to return undefined in the case where Notifications are not defined, and a boolean for whether you have permissions for the other case. I would split the function to see if Notifications exist into a helper, and call it throughout the functions in this file.

Reply:
Removed this function since it's not actually being called